socs/agents/generator/agent.py

```diff
@@ -424,7 +424,13 @@ def acq(self, session, params=None):
                     self.log.info('Trying to reconnect.')
                     continue
 
-                self.agent.publish_to_feed('generator', data)
+                for field, val in data['data'].items():
+                    _data = {
+                        'timestamp': current_time,
+                        'block_name': field,
+                        'data': {field: val}
```

Comment:
Just curious, does `val` return None here if the generator can't provide the data?

Reply:
previously, the field that couldn't return data would just not be included in the data block, which is what caused errors when trying to publish.

Reply:
right, I see it now, the data block isn't updated because reading the register returns None on lines 169-170. makes sense!